website/docs/guides/collections.md

```diff
@@ -0,0 +1,336 @@
+---
+id: collections
+title: Streams and Subscriptions
+---
+
+Effection ships with a powerful library for working with streams of values.
+This allows you to create complex systems where evented code, stateful code and
+concurrent asynchronous code all are working together seamlessly.
+
+If you're familiar with reactive programming libraries such as [Rx.js][rx],
+then this functionality should feel familiar. It is also somewhat similar to
+[asynchronous iterators][].
+
+## Queue
+
+Before we look at Subscriptions and Streams, let's look at another powerful
+concept: the Queue. Effection ships with a `createQueue` functions which returns
+a `Queue`. It can be used like this:
+
+``` javascript
+import { main, createQueue } from 'effection';
+
+main(function*() {
+  let queue = createQueue();
+
+  queue.send('hello');
+  queue.send('world');
+
+  console.log(yield queue.expect()); // logs 'hello'
+  console.log(yield queue.expect()); // logs 'world'
+});
+```
+
+As you can see, we can push values into the queue using `send`, and then we can
+fetch values from the queue using `expect`.
```

Comment:
Hmm, I've actually used `Queue` a lot, I disagree that it's so low level.

This section was pretty challenging to write. Especially the order of things. You can't really talk about subscriptions without talking about some way of creating them. There are currently two principal ways of creating a subscription: from a stream, or from a queue. The former requires us to talk about streams first, but we can't really talk about streams without also mentioning how to consume them... which requires a subscription. So that's how I landed on talking about queues first.